src/Microsoft.AspNetCore.Mvc.RazorPages/Internal/PageActionInvoker.cs

```diff
@@ -137,17 +135,21 @@ protected override Task InvokeResultAsync(IActionResult result)
 
         private object CreateInstance()
         {
-            if (HasPageModel)
+            if (_actionDescriptor.ModelTypeInfo != _actionDescriptor.PageTypeInfo)
             {
-                // Since this is a PageModel, we need to activate it, and then run a handler method on the model.
+                // The page has a model. Initialize it.
```

Comment:
So what's the problem here? Are we now creating POCO model types through the model factory? That was never intended.